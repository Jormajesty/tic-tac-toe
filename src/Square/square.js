import React from "react";
import ReactDOM from "react-dom";

class Square extends React.Component {
	}

export default Square;