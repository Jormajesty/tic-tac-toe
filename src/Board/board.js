import React from 'react'
import Square from '../Square/square'

class Board extends React.Component {
	render() {
	    return <div className="board">testtesttest</div>;
	  }
}
export default Board;