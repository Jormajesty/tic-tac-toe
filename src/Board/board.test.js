import React from "react";
import { shallow, configure, mount } from "enzyme";
import { expect } from 'chai';
import Adapter from "enzyme-adapter-react-16";
import Board from "./Board";
import Square from "../Square/square";

configure({ adapter: new Adapter() });

const board = shallow(<Board />);

describe("Board component", () => {
	it('renders without crashing', () => {
	  let squares = Array(9).fill(null);
	  shallow(<Board squares={squares}/>);
	});
	it("Square value should turn to 'X' during inital click", () => {
		let squares = Array(9).fill(null);
		const onClick = jest.fn();
		let boardClick = mount(<Board squares={squares} onClick={onClick}/>);
		let button = boardClick.find('button').first();
		button.simulate('click');
		expect(button.text()).to.equal('X');
  });
	
});