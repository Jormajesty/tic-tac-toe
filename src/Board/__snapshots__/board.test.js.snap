// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering the app renders Board correctly 1`] = `ShallowWrapper {}`;

exports[`when rendering the board renders Board correctly 1`] = `ShallowWrapper {}`;
